etna/transforms/missing_values/imputation.py

```diff
@@ -82,204 +93,155 @@ def __init__(
         ValueError:
             if incorrect strategy given
         """
+        super().__init__(required_features=[in_column])
         self.in_column = in_column
-        self.strategy = ImputerMode(strategy)
+        self.strategy = strategy
         self.window = window
         self.seasonality = seasonality
         self.default_value = default_value
         self.constant_value = constant_value
-        self.fill_value: Optional[float] = None
-        self.nan_timestamps: Optional[List[pd.Timestamp]] = None
+        self._strategy = ImputerMode(strategy)
+        self._fill_value: Optional[Dict[str, float]] = None
+        self._nan_timestamps: Optional[Dict[str, List[pd.Timestamp]]] = None
 
-    def fit(self, df: pd.DataFrame) -> "_OneSegmentTimeSeriesImputerTransform":
-        """
-        Fit preprocess params.
+    def get_regressors_info(self) -> List[str]:
+        """Return the list with regressors created by the transform."""
+        return []
 
-        Parameters
-        ----------
-        df: pd.DataFrame
-            dataframe with series to fit preprocess params with
+    def _fit(self, df: pd.DataFrame):
+        """Fit the transform.
 
-        Returns
-        -------
-        self: _OneSegmentTimeSeriesImputerTransform
-            fitted preprocess
-        """
-        raw_series = df[self.in_column]
-        if np.all(raw_series.isna()):
-            raise ValueError("Series hasn't non NaN values which means it is empty and can't be filled.")
-        series = raw_series[raw_series.first_valid_index() :]
-        self.nan_timestamps = series[series.isna()].index
-        if self.strategy == ImputerMode.constant:
-            self.fill_value = self.constant_value
-        elif self.strategy == ImputerMode.mean:
-            self.fill_value = series.mean()
-        return self
-
-    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
-        """
-        Transform given series.
+        Should be implemented by user.
```

Comment:
I think we can remove this line, same for the `_transform` method

Reply:
Agree, I'll fix that.